File: Daily Resources/Day05/DSA - Prefix and Difference Arrays Solutions/static_range_sum_queries.py
n, q = map(int, input().split())
arr = list(map(int, input().split()))

# Compute prefix sums (1-based indexing)
prefix = [0] * (n + 1)
for i in range(n):
    prefix[i + 1] = prefix[i] + arr[i]

# Process queries
for _ in range(q):
    a, b = map(int, input().split())
    print(prefix[b] - prefix[a - 1])

File: Daily Resources/Day05/DSA - Prefix and Difference Arrays Solutions/static_range_sum_queries.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, q;
    cin >> n >> q;
    
    vector<long long> arr(n);
    vector<long long> prefix(n + 1, 0);  // 1-based prefix sum

    for (int i = 0; i < n; ++i) {
        cin >> arr[i];
        prefix[i + 1] = prefix[i] + arr[i];
    }

    while (q--) {
        int a, b;
        cin >> a >> b;
        cout << prefix[b] - prefix[a - 1] << '\n';
    }

    return 0;
}
